agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -642,7 +643,6 @@ func (s *Server) realHandleStream(streamReq HandleStreamRequest) error {
 			if err := streamSend(replResp); err != nil {
 				return fmt.Errorf("failed to push data for %q: %w", update.CorrelationID, err)
 			}
-			status.TrackSendSuccess()
```

Comment:
No need to track send success; the peer will respond with Ack or Nack.

Reply:
Send errors aside from heartbeats will cause the stream to disconnect and try the next address in the retry loop, so it will be caught in the "disconnected" case if the sends repeatedly fail